pytest_socket.py

```diff
@@ -170,16 +170,52 @@ def host_from_connect_args(args):
         return host_from_address(address)
 
 
+def is_ipaddress(address: str):
+    """
+    Determine if the address is a valid IPv4 address.
+    """
+    try:
+        socket.inet_aton(address)
+        return True
+    except OSError:
+        return False
+
+
+def resolve_hostname(hostname):
+    try:
+        return socket.gethostbyname(hostname)
+    except socket.gaierror:
+        return None
+
+
+def treat_allowed(allowed):
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**nitpick:** ‏the name of this function doesn't currently sit well for me. Naming things is hard - let's get a better function name and param names so that when this function is read, it's clear what's happening.

Reply:
I agree. I've modified that function with better names